test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.cs

```diff
@@ -20,238 +20,435 @@
+                      ""http.url"": ""http://{host}:{port}/""
+                    }
+                  }
+                ]
+                ",
+            new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+
+        var t = (Task)this.GetType().InvokeMember(nameof(this.HttpOutCallsAreCollectedSuccessfullyTracesAndMetricsOldSemanticConventionsAsync), BindingFlags.InvokeMethod, null, this, HttpTestData.GetArgumentsFromTestCaseObject(input).First());
+        await t.ConfigureAwait(false);
+    }
+
+    [Fact]
+    public async Task CheckEnrichmentWhenSampling()
+    {
+        await CheckEnrichment(new AlwaysOffSampler(), false, this.url).ConfigureAwait(false);
+        await CheckEnrichment(new AlwaysOnSampler(), true, this.url).ConfigureAwait(false);
+    }
+
+    private static async Task HttpOutCallsAreCollectedSuccessfullyBodyAsync(
+        string host,
+        int port,
+        HttpTestData.HttpOutTestCase tc,
+        bool enableTracing,
+        bool enableMetrics,
+        HttpSemanticConvention? semanticConvention = null)
     {
         bool enrichWithHttpWebRequestCalled = false;
         bool enrichWithHttpWebResponseCalled = false;
         bool enrichWithHttpRequestMessageCalled = false;
         bool enrichWithHttpResponseMessageCalled = false;
         bool enrichWithExceptionCalled = false;
 
-        using var serverLifeTime = TestHttpServer.RunServer(
-            (ctx) =>
-            {
-                ctx.Response.StatusCode = tc.ResponseCode == 0 ? 200 : tc.ResponseCode;
-                ctx.Response.OutputStream.Close();
-            },
-            out var host,
-            out var port);
+        var testUrl = HttpTestData.NormalizeValues(tc.Url, host, port);
 
-        var processor = new Mock<BaseProcessor<Activity>>();
-        tc.Url = HttpTestData.NormalizeValues(tc.Url, host, port);
+        var meterProviderBuilder = Sdk.CreateMeterProviderBuilder();
 
-        var metrics = new List<Metric>();
+        if (enableMetrics)
+        {
+            meterProviderBuilder.AddHttpClientInstrumentation();
 
-        var meterProvider = Sdk.CreateMeterProviderBuilder()
-            .AddHttpClientInstrumentation()
-            .AddInMemoryExporter(metrics)
-            .Build();
+            if (semanticConvention.HasValue)
+            {
+                meterProviderBuilder.ConfigureServices(s =>
+                {
+                    s.AddSingleton<IConfiguration>(new ConfigurationBuilder()
+                        .AddInMemoryCollection(
+                        new Dictionary<string, string>
+                        {
+                            ["OTEL_SEMCONV_STABILITY_OPT_IN"] = semanticConvention == HttpSemanticConvention.Dupe
+                                ? "http/dup"
+                                : semanticConvention == HttpSemanticConvention.New
+                                    ? "http"
+                                    : "_invalid",
```

Comment:
nit: I've been using `null` for the default behavior.
I only bring it up because i don't want someone to confuse an "invalid" test case with what is actually the default.

Reply:
I pushed a couple tweaks to reduce the code duplication and removed the "_invalid" part.